<a href="https://colab.research.google.com/github/Koks-creator/Real-Time-People-Counting-and-Occupancy-Prediction-System/blob/main/yolov11.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/gdrive')
!ln -s /content/gdrive/My\ Drive/ /mydrive

Mounted at /content/gdrive


In [ ]:
!unzip /mydrive/yoloTrain/train_data.zip -d /content

Archive:  /mydrive/yoloTrain/train_data.zip
   creating: /content/train_data/
   creating: /content/train_data/images/
   creating: /content/train_data/images/train/
  inflating: /content/train_data/images/train/16-24URBANPEDBIKE02988.jpg  
  inflating: /content/train_data/images/train/17walkspan-cnd-articleLarge.jpg  
  inflating: /content/train_data/images/train/abekcalete_487.jpg  
  inflating: /content/train_data/images/train/abekcalete_488.jpg  
  inflating: /content/train_data/images/train/abekcalete_489.jpg  
  inflating: /content/train_data/images/train/abekcalete_490.jpg  
  inflating: /content/train_data/images/train/abekcalete_491.jpg  
  inflating: /content/train_data/images/train/abekcalete_492.jpg  
  inflating: /content/train_data/images/train/abekcalete_493.jpg  
  inflating: /content/train_data/images/train/abekcalete_494.jpg  
  inflating: /content/train_data/images/train/abekcalete_495.jpg  
  inflating: /content/train_data/images/train/abekcalete_496.jpg  
  inflati

In [ ]:
!pip install ultralytics==8.3.89 sahi==0.11.22

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 932.6/932.6 kB 31.9 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 114.2/114.2 kB 11.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.5/62.5 MB 13.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 115.9/115.9 kB 14.3 MB/s eta 0:00:00
  Attempting uninstall: opencv-python
    Found existing installation: opencv-python 4.13.0.92
    Uninstalling opencv-python-4.13.0.92:
      Successfully uninstalled opencv-python-4.13.0.92


In [ ]:
!pip freeze | grep -e ultralytics -e sahi -e torch

sahi==0.11.22
torch==2.10.0+cu128
torchao==0.10.0
torchaudio==2.10.0+cu128
torchcodec==0.10.0+cu128
torchdata==0.11.0
torchsummary==1.5.1
torchtune==0.6.1
torchvision==0.25.0+cu128
ultralytics==8.3.89
ultralytics-thop==2.0.18


In [ ]:
from ultralytics import YOLO
from sahi.utils.ultralytics import download_yolo11n_model
import yaml
import os
import torch
import shutil

Creating new Ultralytics Settings v0.0.6 file ✅ 
View Ultralytics Settings with 'yolo settings' or at '/root/.config/Ultralytics/settings.json'
Update Settings with 'yolo settings key=value', i.e. 'yolo settings runs_dir=path/to/dir'. For help see https://docs.ultralytics.com/quickstart/#ultralytics-settings.


In [ ]:
if torch.cuda.is_available():
    device_count = torch.cuda.device_count()
    device_name = torch.cuda.get_device_name(0) if device_count > 0 else "None"
    print(f"Found {device_count} GPU devices.")
    print(f"Active device: {device_name}")
    device = 0  # Użyj pierwszego GPU
else:
    print("GPU not found, using CPU.")
    device = 'cpu'

Found 1 GPU devices.
Active device: Tesla T4


In [ ]:
DATASET_PATH = "/content/train_data"
CLASS_NAMES = ["person"]
MODEL_DIR = "./models"
# MODEL_NAME = "yolo11m.pt"
MODEL_NAME = "yolo11s.pt"
MODEL_PATH = os.path.join(MODEL_DIR, MODEL_NAME)
BATCH_SIZE = 16
WORKERS = 8
CONFIG_YAML_FILE = "./config/data.yaml"

In [ ]:
os.makedirs(MODEL_DIR, exist_ok=True)

In [ ]:
data_yaml = {
    'path': DATASET_PATH,
    'train': 'images/train',
    'val': 'images/val',
    'names': {i: name for i, name in enumerate(CLASS_NAMES)},
    'nc': len(CLASS_NAMES)
}

os.makedirs("config", exist_ok=True)
with open(CONFIG_YAML_FILE, 'w') as f:
    yaml.dump(data_yaml, f, sort_keys=False)

print(f"Config file created: {CONFIG_YAML_FILE}")

Config file created: ./config/data.yaml


In [ ]:
print(f"Loading model from {MODEL_PATH}...")
model = YOLO(MODEL_PATH)

Loading model from ./models/yolo11s.pt...


100%|██████████| 18.4M/18.4M [00:00<00:00, 115MB/s] 


In [ ]:
MODEL_PATH

'./models/yolo11m.pt'

In [ ]:
# test vanila params
results = model.train(
    data=CONFIG_YAML_FILE,
    epochs=100,
    imgsz=640,
    batch=BATCH_SIZE,
    project="runs",
    name="yolov11_person",
    save=True,
    device=device,
    workers=WORKERS
)

New https://pypi.org/project/ultralytics/8.4.21 available 😃 Update with 'pip install -U ultralytics'
Ultralytics 8.3.89 🚀 Python-3.12.12 torch-2.10.0+cu128 CUDA:0 (Tesla T4, 14913MiB)
engine/trainer: task=detect, mode=train, model=./models/yolo11m.pt, data=./config/data.yaml, epochs=100, time=None, patience=100, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=0, workers=8, project=runs, name=yolov11_person, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, sav

100%|██████████| 755k/755k [00:00<00:00, 107MB/s]


Overriding model.yaml nc=80 with nc=1

                   from  n    params  module                                       arguments                     
  0                  -1  1      1856  ultralytics.nn.modules.conv.Conv             [3, 64, 3, 2]                 
  1                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  2                  -1  1    111872  ultralytics.nn.modules.block.C3k2            [128, 256, 1, True, 0.25]     
  3                  -1  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]              
  4                  -1  1    444928  ultralytics.nn.modules.block.C3k2            [256, 512, 1, True, 0.25]     
  5                  -1  1   2360320  ultralytics.nn.modules.conv.Conv             [512, 512, 3, 2]              
  6                  -1  1   1380352  ultralytics.nn.modules.block.C3k2            [512, 512, 1, True]           
  7                  -1  1   2360320  ultralytics

100%|██████████| 5.35M/5.35M [00:00<00:00, 417MB/s]


AMP: checks passed ✅


train: Scanning /content/train_data/labels/train... 521 images, 5 backgrounds, 1 corrupt: 100%|██████████| 526/526 [00:00<00:00, 1387.72it/s]

train: WARNING ⚠️ /content/train_data/images/train/ajajaj_520.jpg: 1 duplicate labels removed
train: WARNING ⚠️ /content/train_data/images/train/gunsondupaxd_0_239.png: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.3796]
train: New cache created: /content/train_data/labels/train.cache


albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))


val: Scanning /content/train_data/labels/val... 45 images, 0 backgrounds, 0 corrupt: 100%|██████████| 45/45 [00:00<00:00, 1578.53it/s]

val: New cache created: /content/train_data/labels/val.cache


Plotting labels to runs/yolov11_person/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.002, momentum=0.9) with parameter groups 106 weight(decay=0.0), 113 weight(decay=0.0005), 112 bias(decay=0.0)
TensorBoard: model graph visualization added ✅
Image sizes 640 train, 640 val
Using 2 dataloader workers
Logging results to runs/yolov11_person
Starting training for 100 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


  0%|          | 0/33 [00:00<?, ?it/s]Process Process-4:
Traceback (most recent call last):
  File "/usr/lib/python3.12/multiprocessing/process.py", line 317, in _bootstrap
    util._exit_function()
  File "/usr/lib/python3.12/multiprocessing/util.py", line 363, in _exit_function
    _run_finalizers()
  File "/usr/lib/python3.12/multiprocessing/util.py", line 303, in _run_finalizers
    finalizer()
  File "/usr/lib/python3.12/multiprocessing/util.py", line 227, in __call__
    res = self._callback(*self._args, **self._kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/multiprocessing/queues.py", line 219, in _finalize_join
    thread.join()
  File "/usr/lib/python3.12/threading.py", line 1149, in join
    self._wait_for_tstate_lock()
  File "/usr/lib/python3.12/threading.py", line 1169, in _wait_for_tstate_lock
    if lock.acquire(block, timeout):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
KeyboardInterrupt
  0%|          | 0/33 [00:02<?, ?it/s]


RuntimeError: DataLoader worker (pid 3635) exited unexpectedly with exit code 1. Details are lost due to multiprocessing. Rerunning with num_workers=0 may give better error trace.

In [ ]:
# import torch, gc

# del model
# gc.collect()
# torch.cuda.empty_cache()

In [ ]:
results = model.train(
  data=CONFIG_YAML_FILE,
  # Trening
  epochs=150,
  imgsz=768,
  batch=BATCH_SIZE,
  patience=40,
  project="runs",
  name="yolov11_person_small_dataset",
  pretrained=True,
  optimizer='AdamW',
  lr0=0.001,
  lrf=0.01,
  momentum=0.937,
  weight_decay=0.001,
  warmup_epochs=5,
  cos_lr=True,
  augment=True,
  # Augmentation
  mosaic=0.7,
  mixup=0.05,
  copy_paste=0.2,
  degrees=10.0,
  scale=0.5,
  translate=0.15,
  fliplr=0.5,
  flipud=0.0,
  hsv_h=0.015,
  hsv_s=0.7,
  hsv_v=0.5,
  perspective=0.0005,
  shear=1.0,
  # Loss
  cls=0.5,
  box=8,
  dfl=1.5,
  conf=0.25,
  iou=0.75,
  max_det=50,
  save=True,
  save_period=10,
  device=device,
  workers=WORKERS,
  # Performance
  amp=True,
  profile=False,
  verbose=True,
  close_mosaic=15,
  cache=True,
  dropout=0.1
)

New https://pypi.org/project/ultralytics/8.4.24 available 😃 Update with 'pip install -U ultralytics'
Ultralytics 8.3.89 🚀 Python-3.12.12 torch-2.10.0+cu128 CUDA:0 (Tesla T4, 14913MiB)
engine/trainer: task=detect, mode=train, model=./models/yolo11s.pt, data=./config/data.yaml, epochs=150, time=None, patience=40, batch=16, imgsz=768, save=True, save_period=10, cache=True, device=0, workers=8, project=runs, name=yolov11_person_small_dataset, exist_ok=False, pretrained=True, optimizer=AdamW, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=True, close_mosaic=15, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.1, val=True, split=val, save_json=False, save_hybrid=False, conf=0.25, iou=0.75, max_det=50, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=True, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show

100%|██████████| 755k/755k [00:00<00:00, 28.3MB/s]


Overriding model.yaml nc=80 with nc=1

                   from  n    params  module                                       arguments                     
  0                  -1  1       928  ultralytics.nn.modules.conv.Conv             [3, 32, 3, 2]                 
  1                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                
  2                  -1  1     26080  ultralytics.nn.modules.block.C3k2            [64, 128, 1, False, 0.25]     
  3                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]              
  4                  -1  1    103360  ultralytics.nn.modules.block.C3k2            [128, 256, 1, False, 0.25]    
  5                  -1  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]              
  6                  -1  1    346112  ultralytics.nn.modules.block.C3k2            [256, 256, 1, True]           
  7                  -1  1   1180672  ultralytics

100%|██████████| 5.35M/5.35M [00:00<00:00, 116MB/s]


AMP: checks passed ✅


train: Scanning /content/train_data/labels/train... 521 images, 5 backgrounds, 1 corrupt: 100%|██████████| 526/526 [00:00<00:00, 1172.30it/s]

train: WARNING ⚠️ /content/train_data/images/train/ajajaj_520.jpg: 1 duplicate labels removed
train: WARNING ⚠️ /content/train_data/images/train/gunsondupaxd_0_239.png: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.3796]
train: New cache created: /content/train_data/labels/train.cache


WARNING ⚠️ cache='ram' may produce non-deterministic training results. Consider cache='disk' as a deterministic alternative if your disk space allows.


train: Caching images (0.5GB RAM): 100%|██████████| 525/525 [00:04<00:00, 125.08it/s]


albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))


val: Scanning /content/train_data/labels/val... 45 images, 0 backgrounds, 0 corrupt: 100%|██████████| 45/45 [00:00<00:00, 1513.78it/s]

val: New cache created: /content/train_data/labels/val.cache
WARNING ⚠️ cache='ram' may produce non-deterministic training results. Consider cache='disk' as a deterministic alternative if your disk space allows.



val: Caching images (0.0GB RAM): 100%|██████████| 45/45 [00:00<00:00, 132.24it/s]


Plotting labels to runs/yolov11_person_small_dataset/labels.jpg... 
optimizer: AdamW(lr=0.001, momentum=0.937) with parameter groups 81 weight(decay=0.0), 88 weight(decay=0.001), 87 bias(decay=0.0)
TensorBoard: model graph visualization added ✅
Image sizes 768 train, 768 val
Using 2 dataloader workers
Logging results to runs/yolov11_person_small_dataset
Starting training for 150 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      1/150      9.22G      1.992      1.979      1.366        365        768: 100%|██████████| 33/33 [00:38<00:00,  1.16s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:08<00:00,  4.14s/it]

                   all         45        721       0.36      0.359      0.312      0.113



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      2/150      10.1G      1.897      1.312      1.301        321        768: 100%|██████████| 33/33 [00:15<00:00,  2.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.89it/s]

                   all         45        721      0.638      0.503      0.575      0.281



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      3/150      10.1G       1.85      1.277      1.291        301        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.81it/s]

                   all         45        721        0.6      0.588      0.623      0.327



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      4/150      10.1G      1.828      1.232      1.279        319        768: 100%|██████████| 33/33 [00:14<00:00,  2.21it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.85it/s]

                   all         45        721      0.664      0.592      0.657      0.316



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      5/150      10.1G      1.858      1.239       1.28        333        768: 100%|██████████| 33/33 [00:14<00:00,  2.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.19it/s]

                   all         45        721      0.592      0.578      0.614       0.25



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      6/150      10.1G      1.833      1.252      1.292        340        768: 100%|██████████| 33/33 [00:15<00:00,  2.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.23it/s]

                   all         45        721      0.635      0.609      0.665      0.341



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      7/150      10.1G      1.811      1.203      1.284        385        768: 100%|██████████| 33/33 [00:15<00:00,  2.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.11it/s]

                   all         45        721      0.675       0.56      0.631      0.321



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      8/150      10.1G      1.803      1.174      1.272        336        768: 100%|██████████| 33/33 [00:15<00:00,  2.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.12it/s]

                   all         45        721      0.694      0.634      0.708      0.367



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      9/150      10.1G      1.789       1.14      1.262        562        768: 100%|██████████| 33/33 [00:15<00:00,  2.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.25it/s]

                   all         45        721        0.7      0.641      0.699      0.359



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     10/150      10.1G      1.786      1.145      1.272        198        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.12it/s]

                   all         45        721      0.678      0.624      0.692      0.362



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     11/150      10.1G      1.762      1.137      1.277        265        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.14it/s]

                   all         45        721      0.715      0.635       0.71      0.377



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     12/150      10.1G      1.788       1.14      1.258        379        768: 100%|██████████| 33/33 [00:15<00:00,  2.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:01<00:00,  1.89it/s]

                   all         45        721      0.721      0.653      0.729       0.38



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     13/150      10.1G      1.764      1.144      1.249        434        768: 100%|██████████| 33/33 [00:15<00:00,  2.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.58it/s]

                   all         45        721      0.742       0.61      0.715      0.404



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     14/150      10.1G      1.726      1.099      1.238        795        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.90it/s]

                   all         45        721       0.72      0.606      0.695      0.383



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     15/150      10.1G      1.751      1.104      1.245        465        768: 100%|██████████| 33/33 [00:15<00:00,  2.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.91it/s]

                   all         45        721      0.708       0.68      0.728      0.418



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     16/150      10.1G      1.721      1.094      1.228        492        768: 100%|██████████| 33/33 [00:15<00:00,  2.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.33it/s]

                   all         45        721       0.71      0.645       0.72      0.409



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     17/150      10.1G      1.716      1.097      1.243        200        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.95it/s]

                   all         45        721      0.737      0.618      0.715      0.385



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     18/150      10.1G      1.724      1.102      1.227        413        768: 100%|██████████| 33/33 [00:15<00:00,  2.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.75it/s]

                   all         45        721      0.724      0.671      0.734       0.36



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     19/150      10.1G      1.655      1.068      1.217        260        768: 100%|██████████| 33/33 [00:15<00:00,  2.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.03it/s]

                   all         45        721      0.709      0.653      0.727      0.351



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     20/150      10.1G      1.664       1.04      1.216        354        768: 100%|██████████| 33/33 [00:15<00:00,  2.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.35it/s]

                   all         45        721      0.732      0.701       0.75      0.395



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     21/150      10.1G      1.645      1.013      1.207        428        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.91it/s]

                   all         45        721      0.762       0.67      0.759      0.411



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     22/150      10.1G      1.659      1.021      1.204        269        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.27it/s]


                   all         45        721      0.761      0.669      0.753      0.423

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     23/150      10.1G      1.632      1.017      1.201        411        768: 100%|██████████| 33/33 [00:15<00:00,  2.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.49it/s]

                   all         45        721      0.744      0.645      0.735      0.436



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     24/150      10.1G      1.615     0.9883      1.178        238        768: 100%|██████████| 33/33 [00:15<00:00,  2.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:01<00:00,  2.00it/s]

                   all         45        721      0.739      0.702      0.761      0.421



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     25/150      10.1G      1.635     0.9899      1.201        379        768: 100%|██████████| 33/33 [00:14<00:00,  2.23it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.44it/s]

                   all         45        721      0.763      0.662      0.738      0.436



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     26/150      10.1G      1.644      0.996      1.196        256        768: 100%|██████████| 33/33 [00:15<00:00,  2.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.93it/s]


                   all         45        721      0.718      0.661      0.724      0.416

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     27/150      10.1G      1.602     0.9818      1.176        327        768: 100%|██████████| 33/33 [00:15<00:00,  2.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.56it/s]

                   all         45        721      0.713      0.665      0.724      0.424



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     28/150      10.1G      1.639      1.001      1.189        312        768: 100%|██████████| 33/33 [00:14<00:00,  2.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.21it/s]

                   all         45        721       0.72      0.653      0.724      0.424



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     29/150      10.1G      1.598     0.9888      1.185        175        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.22it/s]

                   all         45        721      0.743      0.666      0.735      0.416



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     30/150      10.1G      1.588     0.9667      1.175        307        768: 100%|██████████| 33/33 [00:15<00:00,  2.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.81it/s]

                   all         45        721      0.749      0.653      0.739      0.415



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     31/150      10.1G      1.583     0.9602      1.165        252        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.68it/s]

                   all         45        721      0.699      0.686      0.734      0.419



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     32/150      10.1G      1.564     0.9447      1.173        356        768: 100%|██████████| 33/33 [00:15<00:00,  2.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.22it/s]

                   all         45        721      0.684      0.695      0.743      0.404



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     33/150      10.1G      1.553     0.9273       1.15        261        768: 100%|██████████| 33/33 [00:15<00:00,  2.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.82it/s]

                   all         45        721      0.745      0.679      0.743      0.429



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     34/150      10.1G      1.587      0.933      1.172        209        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.33it/s]

                   all         45        721      0.777      0.668      0.755      0.407



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     35/150      10.1G      1.564     0.9158      1.153        412        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.83it/s]

                   all         45        721      0.764      0.648      0.743      0.436



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     36/150      10.1G      1.582      0.928      1.162        377        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.33it/s]

                   all         45        721      0.739      0.653      0.742      0.405



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     37/150        11G      1.549     0.9213      1.156        444        768: 100%|██████████| 33/33 [00:14<00:00,  2.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.09it/s]

                   all         45        721      0.764      0.655      0.752      0.441



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     38/150        11G      1.553     0.9214       1.15        263        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.10it/s]

                   all         45        721      0.745      0.611      0.711      0.425



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     39/150        11G      1.575     0.9293      1.155        255        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.53it/s]

                   all         45        721      0.739      0.707      0.765      0.452



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     40/150        11G      1.567     0.9399      1.155        333        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.83it/s]

                   all         45        721      0.739       0.69      0.765      0.433



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     41/150        11G      1.524     0.9191      1.147        161        768: 100%|██████████| 33/33 [00:15<00:00,  2.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.33it/s]

                   all         45        721      0.738      0.667       0.75      0.437



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     42/150        11G       1.53     0.9034      1.141        291        768: 100%|██████████| 33/33 [00:14<00:00,  2.21it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.63it/s]

                   all         45        721      0.789      0.664      0.767       0.42



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     43/150        11G      1.516     0.8963      1.136        436        768: 100%|██████████| 33/33 [00:15<00:00,  2.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.19it/s]

                   all         45        721      0.739      0.679      0.764      0.431



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     44/150        11G      1.511     0.8794       1.13        349        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.13it/s]

                   all         45        721      0.764      0.714      0.783      0.461



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     45/150        11G      1.521     0.9015      1.143        334        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.86it/s]

                   all         45        721      0.765      0.645      0.732      0.445



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     46/150        11G      1.504      0.874      1.133        246        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.32it/s]

                   all         45        721      0.781      0.634       0.74      0.439



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     47/150        11G       1.48     0.8556      1.126        348        768: 100%|██████████| 33/33 [00:14<00:00,  2.21it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.39it/s]

                   all         45        721      0.728      0.693      0.748      0.426



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     48/150        11G      1.482      0.848      1.128        637        768: 100%|██████████| 33/33 [00:14<00:00,  2.21it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.82it/s]

                   all         45        721        0.8      0.666      0.766      0.456



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     49/150        11G      1.499     0.8397      1.112        329        768: 100%|██████████| 33/33 [00:14<00:00,  2.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.36it/s]

                   all         45        721      0.806       0.65      0.758      0.465



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     50/150        11G      1.483     0.8376      1.119        561        768: 100%|██████████| 33/33 [00:15<00:00,  2.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.11it/s]

                   all         45        721      0.783       0.63      0.738      0.444



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     51/150        11G      1.456     0.8329      1.111        156        768: 100%|██████████| 33/33 [00:14<00:00,  2.22it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.76it/s]

                   all         45        721      0.721      0.678      0.743      0.442



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     52/150        11G      1.467     0.8562      1.114        295        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.16it/s]

                   all         45        721      0.794      0.675      0.776       0.46



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     53/150        11G      1.471     0.8428      1.111        228        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.18it/s]

                   all         45        721      0.705      0.665      0.731      0.426



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     54/150        11G      1.487     0.8551      1.112        322        768: 100%|██████████| 33/33 [00:15<00:00,  2.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.13it/s]

                   all         45        721      0.771      0.669      0.752      0.432



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     55/150        11G      1.437     0.8225      1.106        242        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.07it/s]

                   all         45        721       0.73      0.671       0.74      0.451



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     56/150        11G      1.466      0.823      1.096        435        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.33it/s]

                   all         45        721      0.723      0.717       0.77      0.464



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     57/150        11G      1.465     0.8297       1.11        245        768: 100%|██████████| 33/33 [00:15<00:00,  2.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.29it/s]

                   all         45        721      0.743      0.692      0.747      0.446



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     58/150        11G      1.478     0.8465      1.122        280        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.89it/s]

                   all         45        721      0.766      0.673      0.761      0.461



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     59/150        11G      1.482     0.8333      1.095        419        768: 100%|██████████| 33/33 [00:15<00:00,  2.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.28it/s]

                   all         45        721      0.764      0.684      0.756      0.445



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     60/150        11G      1.446     0.8186      1.098        296        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.56it/s]

                   all         45        721       0.77      0.656      0.749      0.454



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     61/150        11G      1.452     0.8065      1.103        269        768: 100%|██████████| 33/33 [00:15<00:00,  2.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:01<00:00,  1.91it/s]

                   all         45        721      0.762      0.637       0.74      0.448



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     62/150        11G      1.452     0.8256      1.099        468        768: 100%|██████████| 33/33 [00:14<00:00,  2.22it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.85it/s]

                   all         45        721      0.769      0.669      0.769      0.461



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     63/150        11G      1.416     0.7975      1.096        291        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.15it/s]

                   all         45        721      0.753      0.723      0.783      0.475



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     64/150        11G      1.404     0.7904      1.081        270        768: 100%|██████████| 33/33 [00:15<00:00,  2.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.20it/s]

                   all         45        721      0.756       0.72      0.774      0.464



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     65/150        11G      1.391     0.7852      1.074        284        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.77it/s]

                   all         45        721      0.766      0.674      0.768      0.456



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     66/150        11G      1.447     0.7976      1.087        344        768: 100%|██████████| 33/33 [00:15<00:00,  2.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.13it/s]

                   all         45        721      0.766      0.675      0.769      0.446



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     67/150        11G       1.42     0.7928      1.082        308        768: 100%|██████████| 33/33 [00:15<00:00,  2.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.15it/s]

                   all         45        721      0.754      0.667      0.755      0.405



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     68/150        11G      1.446     0.7988      1.088        294        768: 100%|██████████| 33/33 [00:15<00:00,  2.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.18it/s]

                   all         45        721       0.72      0.685      0.756      0.441



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     69/150        11G      1.391      0.776      1.068        280        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.00it/s]

                   all         45        721      0.752      0.699       0.77       0.46



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     70/150        11G      1.386     0.7871      1.076        285        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.15it/s]

                   all         45        721      0.754      0.699      0.762      0.441



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     71/150        12G       1.38     0.7689       1.07        289        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.53it/s]

                   all         45        721      0.786      0.682      0.773      0.464



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     72/150        12G      1.393     0.7641      1.075        232        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.40it/s]

                   all         45        721      0.765      0.705      0.773      0.476



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     73/150        12G      1.337     0.7386      1.077        316        768: 100%|██████████| 33/33 [00:14<00:00,  2.21it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.92it/s]

                   all         45        721      0.737       0.71      0.764      0.478



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     74/150        12G      1.363      0.749      1.068        335        768: 100%|██████████| 33/33 [00:15<00:00,  2.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.02it/s]

                   all         45        721      0.793      0.662      0.756      0.465



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     75/150        12G      1.323     0.7287      1.058        283        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.97it/s]

                   all         45        721      0.777      0.705      0.777      0.467



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     76/150        12G      1.348     0.7466      1.067        211        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.25it/s]

                   all         45        721      0.737      0.717      0.776      0.464



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     77/150        12G      1.327     0.7364      1.063        372        768: 100%|██████████| 33/33 [00:15<00:00,  2.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.29it/s]

                   all         45        721      0.743       0.72      0.773      0.472



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     78/150        12G      1.352     0.7337      1.053        314        768: 100%|██████████| 33/33 [00:15<00:00,  2.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.07it/s]

                   all         45        721       0.78      0.692      0.773       0.47



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     79/150        12G      1.303     0.7209      1.041        341        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.24it/s]

                   all         45        721      0.808      0.698       0.78      0.463



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     80/150        12G      1.315     0.7185      1.053        301        768: 100%|██████████| 33/33 [00:15<00:00,  2.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.27it/s]

                   all         45        721      0.791      0.681      0.777      0.458



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     81/150        12G      1.315     0.7241      1.054        396        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.12it/s]

                   all         45        721      0.788      0.674      0.766      0.476



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     82/150        12G      1.331     0.7254      1.047        232        768: 100%|██████████| 33/33 [00:15<00:00,  2.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.34it/s]

                   all         45        721      0.778       0.67      0.755      0.469



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     83/150        12G      1.309     0.7151      1.043        381        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.06it/s]

                   all         45        721      0.739      0.723       0.78      0.478



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     84/150        12G      1.311     0.7146      1.056        325        768: 100%|██████████| 33/33 [00:15<00:00,  2.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.05it/s]

                   all         45        721       0.81      0.674      0.774      0.477



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     85/150        12G      1.313     0.7161      1.044        446        768: 100%|██████████| 33/33 [00:15<00:00,  2.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.31it/s]

                   all         45        721      0.784      0.707      0.788      0.477



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     86/150        12G      1.291     0.6999      1.043        445        768: 100%|██████████| 33/33 [00:14<00:00,  2.21it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.81it/s]

                   all         45        721      0.791       0.66      0.761       0.47



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     87/150        12G      1.303      0.717      1.064        328        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.02it/s]

                   all         45        721      0.752      0.723      0.775      0.482



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     88/150        12G      1.308     0.7103      1.043        304        768: 100%|██████████| 33/33 [00:15<00:00,  2.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.84it/s]

                   all         45        721      0.804      0.671      0.771      0.472



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     89/150        12G      1.296     0.7006      1.039        281        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.79it/s]

                   all         45        721      0.749      0.702      0.767      0.465



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     90/150        12G      1.269     0.6846      1.034        208        768: 100%|██████████| 33/33 [00:15<00:00,  2.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.07it/s]

                   all         45        721      0.722      0.723      0.766      0.474



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     91/150        12G      1.255      0.686      1.036        318        768: 100%|██████████| 33/33 [00:14<00:00,  2.21it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.04it/s]

                   all         45        721      0.764      0.706      0.772      0.474



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     92/150        12G      1.246     0.6673      1.026        507        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.42it/s]

                   all         45        721      0.777      0.677      0.756      0.465



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     93/150        12G      1.234     0.6689      1.023        280        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.19it/s]

                   all         45        721       0.79        0.7      0.776      0.466



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     94/150        12G      1.301      0.705      1.033        195        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.39it/s]

                   all         45        721      0.775       0.71      0.777      0.485



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     95/150        12G      1.265     0.6827       1.03        349        768: 100%|██████████| 33/33 [00:15<00:00,  2.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.08it/s]

                   all         45        721       0.74       0.73      0.772       0.47



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     96/150        12G      1.289     0.7035      1.046        270        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.01it/s]

                   all         45        721        0.8      0.669      0.773      0.474



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     97/150        12G      1.257     0.6946      1.035        291        768: 100%|██████████| 33/33 [00:14<00:00,  2.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.61it/s]

                   all         45        721      0.732      0.717      0.766       0.48



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     98/150        12G       1.25     0.6698      1.028        191        768: 100%|██████████| 33/33 [00:15<00:00,  2.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.08it/s]

                   all         45        721      0.788      0.697      0.779      0.485



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     99/150        12G       1.25     0.6735      1.024        346        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.43it/s]

                   all         45        721      0.785      0.693      0.775      0.476



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    100/150        12G      1.253     0.6585       1.02        346        768: 100%|██████████| 33/33 [00:15<00:00,  2.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.13it/s]

                   all         45        721      0.763      0.725      0.781      0.479



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    101/150        12G      1.247      0.664       1.02        285        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.08it/s]

                   all         45        721      0.782       0.68      0.765      0.478



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    102/150        12G      1.244     0.6666      1.026        297        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.93it/s]

                   all         45        721      0.787      0.691      0.766      0.483



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    103/150        12G      1.245     0.6582       1.02        344        768: 100%|██████████| 33/33 [00:15<00:00,  2.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.09it/s]

                   all         45        721      0.811      0.684      0.777      0.483



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    104/150        12G      1.245     0.6679      1.019        228        768: 100%|██████████| 33/33 [00:15<00:00,  2.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.78it/s]

                   all         45        721      0.765      0.698      0.774      0.459



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    105/150        12G      1.199     0.6451      1.009        331        768: 100%|██████████| 33/33 [00:15<00:00,  2.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.31it/s]

                   all         45        721      0.767      0.715      0.778      0.487



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    106/150        12G      1.196     0.6352      1.004        392        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.93it/s]

                   all         45        721      0.788      0.703       0.78      0.478



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    107/150        12G      1.202     0.6313      1.003        375        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.25it/s]

                   all         45        721      0.769      0.703      0.776      0.487



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    108/150        12G      1.218     0.6588      1.018        222        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:01<00:00,  1.70it/s]

                   all         45        721      0.758      0.704      0.772      0.477



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    109/150        12G      1.197     0.6326     0.9996        229        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.09it/s]

                   all         45        721      0.759      0.699      0.767      0.481



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    110/150        12G      1.202     0.6459      1.012        269        768: 100%|██████████| 33/33 [00:15<00:00,  2.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.52it/s]

                   all         45        721      0.778        0.7      0.782      0.487



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    111/150        12G      1.193     0.6289      1.007        244        768: 100%|██████████| 33/33 [00:15<00:00,  2.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.24it/s]

                   all         45        721      0.762      0.698      0.767      0.473



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    112/150        12G       1.23     0.6569       1.02        395        768: 100%|██████████| 33/33 [00:15<00:00,  2.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.22it/s]

                   all         45        721      0.776      0.706      0.776      0.486



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    113/150        12G      1.181      0.624     0.9999        309        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.93it/s]

                   all         45        721      0.756      0.699      0.764      0.484



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    114/150        12G      1.154     0.6007     0.9836        318        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.97it/s]

                   all         45        721      0.736      0.714      0.765      0.481



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    115/150        12G       1.17     0.6093     0.9905        280        768: 100%|██████████| 33/33 [00:15<00:00,  2.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.17it/s]

                   all         45        721      0.759      0.706      0.765      0.489



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    116/150        12G      1.182     0.6296     0.9964        486        768: 100%|██████████| 33/33 [00:15<00:00,  2.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.17it/s]

                   all         45        721      0.765      0.703      0.777      0.484



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    117/150        12G      1.168     0.6186      1.001        493        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.34it/s]

                   all         45        721      0.778      0.698      0.769      0.485



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    118/150        12G      1.172     0.6144      0.991        499        768: 100%|██████████| 33/33 [00:15<00:00,  2.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.31it/s]

                   all         45        721      0.767      0.689      0.763      0.487



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    119/150        12G       1.16     0.6158     0.9926        346        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:01<00:00,  1.83it/s]

                   all         45        721      0.779      0.685      0.766      0.486



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    120/150        12G      1.181     0.6279      1.004        360        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.49it/s]

                   all         45        721      0.745      0.712      0.767      0.485



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    121/150        12G      1.152     0.6076      0.989        512        768: 100%|██████████| 33/33 [00:15<00:00,  2.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.44it/s]

                   all         45        721      0.767      0.687      0.762      0.485



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    122/150        12G      1.195      0.622     0.9931        449        768: 100%|██████████| 33/33 [00:15<00:00,  2.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.46it/s]

                   all         45        721       0.78      0.692      0.775      0.486



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    123/150        12G       1.13     0.5944     0.9793        236        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.80it/s]

                   all         45        721       0.74      0.728      0.773      0.487



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    124/150        12G      1.151       0.61      0.989        307        768: 100%|██████████| 33/33 [00:15<00:00,  2.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.13it/s]

                   all         45        721      0.766      0.712      0.777      0.489



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    125/150        12G      1.161     0.6206     0.9913        445        768: 100%|██████████| 33/33 [00:15<00:00,  2.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.24it/s]

                   all         45        721      0.789      0.687      0.773       0.49



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    126/150        12G      1.154      0.607     0.9921        530        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.31it/s]

                   all         45        721      0.817      0.663       0.77      0.486



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    127/150        12G      1.122     0.6005     0.9895        155        768: 100%|██████████| 33/33 [00:15<00:00,  2.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.60it/s]

                   all         45        721      0.788      0.678      0.774      0.487



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    128/150        12G      1.147     0.6068     0.9911        337        768: 100%|██████████| 33/33 [00:15<00:00,  2.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.13it/s]

                   all         45        721      0.767      0.696      0.774      0.485



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    129/150        12G      1.152     0.6078     0.9999        325        768: 100%|██████████| 33/33 [00:15<00:00,  2.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.01it/s]

                   all         45        721      0.749      0.725      0.773      0.486



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    130/150        12G      1.173     0.6182      1.005        385        768: 100%|██████████| 33/33 [00:15<00:00,  2.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.27it/s]

                   all         45        721      0.781      0.693      0.767      0.484



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    131/150        12G      1.128     0.6041     0.9869        299        768: 100%|██████████| 33/33 [00:14<00:00,  2.23it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.43it/s]

                   all         45        721      0.777      0.693      0.765      0.483



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    132/150        12G      1.142     0.6039      0.986        222        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.26it/s]

                   all         45        721      0.745      0.717      0.768      0.488



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    133/150        12G      1.132      0.603     0.9897        300        768: 100%|██████████| 33/33 [00:15<00:00,  2.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.61it/s]

                   all         45        721      0.738      0.724       0.77      0.487



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    134/150        12G      1.108     0.5954     0.9824        324        768: 100%|██████████| 33/33 [00:15<00:00,  2.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.53it/s]

                   all         45        721      0.757      0.714      0.771      0.487



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    135/150        12G      1.132     0.5888     0.9778        497        768: 100%|██████████| 33/33 [00:15<00:00,  2.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.08it/s]

                   all         45        721      0.777      0.699      0.771      0.491


Closing dataloader mosaic
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    136/150        12G      1.223     0.6224      1.017         78        768: 100%|██████████| 33/33 [00:16<00:00,  2.05it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.04it/s]

                   all         45        721      0.776      0.695      0.766      0.478



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    137/150        12G      1.209     0.6121      1.018        125        768: 100%|██████████| 33/33 [00:14<00:00,  2.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.86it/s]

                   all         45        721      0.754      0.716      0.764      0.472



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    138/150        12G      1.189     0.6028     0.9965         77        768: 100%|██████████| 33/33 [00:14<00:00,  2.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.22it/s]

                   all         45        721      0.774      0.697      0.765      0.475



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    139/150        12G      1.205     0.6086      1.015        271        768: 100%|██████████| 33/33 [00:14<00:00,  2.25it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.15it/s]

                   all         45        721      0.778      0.707      0.768      0.478



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    140/150        12G      1.196     0.6003      1.007        153        768: 100%|██████████| 33/33 [00:14<00:00,  2.25it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.23it/s]

                   all         45        721      0.774      0.698       0.77      0.484



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    141/150        12G      1.201     0.6064      1.012        105        768: 100%|██████████| 33/33 [00:14<00:00,  2.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  3.98it/s]

                   all         45        721      0.751      0.714      0.768      0.485



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    142/150        12G      1.185     0.6007      1.003        206        768: 100%|██████████| 33/33 [00:14<00:00,  2.27it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.18it/s]

                   all         45        721       0.74      0.725      0.768      0.487



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    143/150        12G      1.191      0.602      1.005        196        768: 100%|██████████| 33/33 [00:14<00:00,  2.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.11it/s]

                   all         45        721      0.742      0.725      0.769      0.487



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    144/150        12G      1.185     0.5956     0.9924        288        768: 100%|██████████| 33/33 [00:14<00:00,  2.25it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.12it/s]

                   all         45        721      0.743      0.723      0.767      0.486



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    145/150        12G      1.177     0.5911     0.9928        246        768: 100%|██████████| 33/33 [00:14<00:00,  2.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  4.03it/s]

                   all         45        721      0.759      0.706      0.765      0.486



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    146/150        12G      1.188     0.5951          1        229        768: 100%|██████████| 33/33 [00:14<00:00,  2.28it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.77it/s]


                   all         45        721       0.77      0.698      0.768      0.484

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    147/150        12G       1.18     0.6009     0.9979        190        768: 100%|██████████| 33/33 [00:14<00:00,  2.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.55it/s]

                   all         45        721      0.753      0.714      0.767      0.486



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    148/150        12G      1.181     0.5967     0.9926        152        768: 100%|██████████| 33/33 [00:14<00:00,  2.26it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.37it/s]

                   all         45        721      0.759       0.71      0.767      0.484



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    149/150        12G      1.183     0.5878     0.9957        161        768: 100%|██████████| 33/33 [00:14<00:00,  2.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.44it/s]

                   all         45        721      0.752      0.714      0.767      0.487



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    150/150        12G      1.188     0.5968     0.9997        178        768: 100%|██████████| 33/33 [00:14<00:00,  2.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:00<00:00,  2.14it/s]

                   all         45        721      0.762      0.703      0.766      0.487



150 epochs completed in 0.687 hours.
Optimizer stripped from runs/yolov11_person_small_dataset/weights/last.pt, 19.2MB
Optimizer stripped from runs/yolov11_person_small_dataset/weights/best.pt, 19.2MB

Validating runs/yolov11_person_small_dataset/weights/best.pt...
Ultralytics 8.3.89 🚀 Python-3.12.12 torch-2.10.0+cu128 CUDA:0 (Tesla T4, 14913MiB)
YOLO11s summary (fused): 100 layers, 9,413,187 parameters, 0 gradients, 21.3 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 2/2 [00:14<00:00,  7.05s/it]


                   all         45        721       0.75      0.678      0.755      0.475
Speed: 0.4ms preprocess, 306.4ms inference, 0.0ms loss, 0.9ms postprocess per image
Results saved to runs/yolov11_person_small_dataset


# YOLOv11 Training Parameters
https://docs.ultralytics.com/modes/train/#train-settings
<br>
https://www.ultralytics.com/glossary/data-augmentation


# YOLOv11 Training Parameters with Practical Guidelines

| Argument | Type | Default | Description | Practical Guideline |
|----------|------|---------|-------------|---------------------|
| `model` | str | None | Specifies the model file for training. Accepts a path to either a .pt pretrained model or a .yaml configuration file. | For most tasks, use a pretrained YOLOv11 model (e.g., 'yolov11n.pt') for transfer learning rather than training from scratch. |
| `data` | str | None | Path to the dataset configuration file (e.g., coco8.yaml) containing dataset-specific parameters. | Create a custom YAML file with your dataset paths and class names; follow the COCO format for best compatibility. |
| `epochs` | int | 100 | Total number of training epochs, each representing a full pass over the dataset. | Use 50-100 for transfer learning; 100-300 for training from scratch; monitor validation metrics to avoid overfitting. |
| `time` | float | None | Maximum training time in hours, overriding the epochs argument. | Useful for time-boxed experiments; set to 24-48 hours for complex datasets when exact epoch count is less important than time constraints. |
| `patience` | int | 100 | Number of epochs to wait without improvement before early stopping. | Set to 20-50 for most tasks; lower (10-15) for quick experiments; higher (50-100) for complex datasets with slow convergence. |
| `batch` | int | 16 | Batch size for training (integer value, -1 for auto-sizing, or fraction for GPU utilization). | Use largest value that fits in memory; 8-16 for 8GB GPU; 32-64 for 24GB+ GPUs; smaller batches (4-8) can improve generalization. |
| `imgsz` | int or list | 640 | Target image size for training, affecting model accuracy and speed. | Use 640-1280 for general detection; larger (1280+) for small objects; smaller (416-512) for speed-critical applications. |
| `save` | bool | True | Enables saving of training checkpoints and final model weights. | Always keep True unless running purely experimental tests where saving is unnecessary. |
| `save_period` | int | -1 | Frequency of saving model checkpoints in epochs. | Set to 10-20 for long trainings to capture model evolution; lower (5) for critical/expensive trainings. |
| `cache` | bool | False | Enables caching of dataset images (True/ram, disk, or False). | Use 'disk' for large datasets; True/ram for small datasets (<5GB) with sufficient RAM; False for most typical cases. |
| `device` | int or str or list | None | Specifies computational device(s) for training. | For single GPU use '0'; multi-GPU use [0,1,2]; CPU-only use 'cpu'; use 'mps' for Apple Silicon. |
| `workers` | int | 8 | Number of worker threads for data loading per GPU. | Set to CPU core count minus 2; reduce to 0-4 if experiencing memory issues; increase to 12-16 for high-speed storage systems. |
| `project` | str | None | Name of the project directory for saving training outputs. | Use descriptive names for specific tasks ("traffic-detection", "crack-segmentation") to organize different research directions. |
| `name` | str | None | Name of the training run, creating a subdirectory within the project folder. | Use versioned names ("v1", "small-batch", "high-lr") that describe the specific training configuration for easy comparison. |
| `exist_ok` | bool | False | Allows overwriting of an existing project/name directory. | Set True during active development; False for production to prevent accidental overwriting of important results. |
| `pretrained` | bool or str | True | Determines whether to start training from a pretrained model. | Use True for most cases; False only when training on very specialized data unlike any pretraining dataset. |
| `optimizer` | str | 'auto' | Choice of optimizer for training. | Use 'SGD' for best generalization; 'Adam' for faster convergence; 'auto' works well for standard cases. |
| `seed` | int | 0 | Sets the random seed for training reproducibility. | Change only when comparing different architectures to ensure fair comparison with identical initialization. |
| `deterministic` | bool | True | Forces deterministic algorithm use, ensuring reproducibility. | Keep True for research and benchmarking; can set False for slight speed improvements in production training. |
| `single_cls` | bool | False | Treats all classes as a single class during training. | Set True only for presence/absence detection problems; False for all multi-class detection tasks. |
| `classes` | list[int] | None | Specifies class IDs to train on. | Use to focus on specific classes in a larger dataset; helpful for transfer learning on a subset of original classes. |
| `rect` | bool | False | Enables rectangular training, optimizing batch composition. | Set True for inference speed optimization; keep False during main training for better data augmentation. |
| `multi_scale` | bool | False | Enables multi-scale training by varying image size during training. | Enable for models that need robust multi-scale detection (traffic, aerial); unnecessary for fixed-distance applications. |
| `cos_lr` | bool | False | Utilizes a cosine learning rate scheduler. | Set True for most trainings; provides smoother LR decay and often better final convergence than step-based schedules. |
| `close_mosaic` | int | 10 | Disables mosaic augmentation in the last N epochs. | Set to 5-10% of total epochs; critical for stabilizing final training phase and improving precision. |
| `resume` | bool | False | Resumes training from the last saved checkpoint. | Use when training was interrupted; ensures consistent continuation of optimization process. |
| `amp` | bool | True | Enables Automatic Mixed Precision training. | Keep True on modern GPUs for memory savings and speed; set False only if experiencing numerical stability issues. |
| `fraction` | float | 1.0 | Fraction of the dataset to use for training. | Use 0.1-0.2 for quick experiments; 0.5 for prototype development; 1.0 for final training. |
| `profile` | bool | False | Enables ONNX and TensorRT speed profiling. | Enable only when planning to deploy with these frameworks to identify potential optimization bottlenecks. |
| `freeze` | int or list | None | Freezes the first N layers or specified layer indices. | Freeze 0-10 backbone layers when fine-tuning; more layers (10-15) for very similar datasets to the pretrained one. |
| `lr0` | float | 0.01 | Initial learning rate. | Use 0.01 for SGD with new models; 0.001-0.005 for fine-tuning; 0.0001-0.0005 for delicate final tuning. |
| `lrf` | float | 0.01 | Final learning rate as a fraction of initial rate. | Set 0.01-0.1 for typical training; lower (0.001-0.01) for longer trainings for smoother decay. |
| `momentum` | float | 0.937 | Momentum factor for SGD or beta1 for Adam. | Keep at 0.9-0.95 for most cases; lower (0.8-0.9) for noisy data; higher (0.95-0.99) for stable, well-behaved loss landscapes. |
| `weight_decay` | float | 0.0005 | L2 regularization factor, penalizing large weights. | Use 0.0005-0.001 for most cases; increase to 0.001-0.005 for small datasets prone to overfitting. |
| `warmup_epochs` | float | 3.0 | Number of epochs for learning rate warmup. | Set to 3-5 for stable training start; higher (5-10) for very large batch sizes or high learning rates. |
| `warmup_momentum` | float | 0.8 | Initial momentum for warmup phase. | Keep at default 0.8 for most cases; rarely needs adjustment. |
| `warmup_bias_lr` | float | 0.1 | Learning rate for bias parameters during warmup. | Keep at default 0.1; higher values (0.2-0.5) can sometimes improve initial detection of rare classes. |
| `box` | float | 7.5 | Weight of the box loss component in the loss function. | Use 5.0-7.5 for accurate localization; higher (8.0-10.0) when precise boundaries are critical; lower (3.0-5.0) to prioritize recall. |
| `cls` | float | 0.5 | Weight of the classification loss in the total loss function. | Increase to 1.0-2.0 to improve classification accuracy; decrease to 0.3-0.5 when localization is more important than classification. |
| `dfl` | float | 1.5 | Weight of the distribution focal loss. | Keep at default 1.5 for most cases; increase to 2.0-3.0 for highly overlapping objects requiring precise distribution. |
| `pose` | float | 12.0 | Weight of the pose loss in models trained for pose estimation. | Only relevant for pose models; increase to 15.0-20.0 for applications requiring very precise keypoint localization. |
| `kobj` | float | 2.0 | Weight of the keypoint objectness loss in pose estimation. | Only for pose models; balance with pose parameter (typically 1:5 to 1:8 ratio) depending on application needs. |
| `nbs` | int | 64 | Nominal batch size for normalization of loss. | Rarely needs changing; adjust only if experiencing unstable gradients with very large or small batch sizes. |
| `overlap_mask` | bool | True | Determines mask merging strategy for overlapping objects. | Keep True for most segmentation tasks; set False only for specialized cases with transparent overlapping objects. |
| `mask_ratio` | int | 4 | Downsample ratio for segmentation masks. | Use 4 for balance of speed and accuracy; lower (2) for fine detail preservation; higher (8) for speed on large masks. |
| `dropout` | float | 0.0 | Dropout rate for regularization in classification. | Set to 0.1-0.3 for classifiers prone to overfitting; keep at 0.0 for most detection tasks unless overfitting is observed. |
| `val` | bool | True | Enables validation during training. | Always keep True to monitor training progress; False only in rare cases where validation would be redundant. |
| `plots` | bool | False | Generates and saves plots of training metrics and examples. | Enable for visualizing training progress and model performance; especially useful for debugging or tuning hyperparameters. |

# YOLO Augmentation Parameters with Practical Guidelines

| Argument | Type | Default | Range | Description | Practical Guideline |
|----------|------|---------|-------|-------------|---------------------|
| `hsv_h` | float | 0.015 | 0.0 - 1.0 | Adjusts the hue of the image by a fraction of the color wheel, introducing color variability. Helps the model generalize across different lighting conditions. | Keep values small (0.015-0.05) for realistic colors; use higher values (0.1-0.2) only when color variation is critical for detection. |
| `hsv_s` | float | 0.7 | 0.0 - 1.0 | Alters the saturation of the image by a fraction, affecting the intensity of colors. Useful for simulating different environmental conditions. | Use higher values (0.5-0.7) for outdoor datasets with varying lighting conditions; lower (0.2-0.4) for controlled indoor environments. |
| `hsv_v` | float | 0.4 | 0.0 - 1.0 | Modifies the value (brightness) of the image by a fraction, helping the model to perform well under various lighting conditions. | Set to 0.3-0.5 for datasets with mixed lighting; increase to 0.6-0.8 for night/low-light detection scenarios. |
| `degrees` | float | 0.0 | 0.0 - 180 | Rotates the image randomly within the specified degree range, improving the model's ability to recognize objects at various orientations. | Use 0-10° for objects with directional properties (road markings, text); 10-30° for orientation-independent objects (people, animals); avoid for horizon-aligned scenes. |
| `translate` | float | 0.1 | 0.0 - 1.0 | Translates the image horizontally and vertically by a fraction of the image size, aiding in learning to detect partially visible objects. | Set to 0.1-0.2 for centered objects; increase to 0.3-0.5 for objects that appear at image edges or are partially visible. |
| `scale` | float | 0.5 | >=0.0 | Scales the image by a gain factor, simulating objects at different distances from the camera. | Use 0.4-0.6 for general detection tasks; higher (0.7-0.9) when object size varies significantly in your dataset. |
| `shear` | float | 0.0 | -180 - +180 | Shears the image by a specified degree, mimicking the effect of objects being viewed from different angles. | Keep low (1.0-3.0) for most tasks; increase to 5.0-10.0 for datasets with varied camera angles; use cautiously as it can distort object shapes. |
| `perspective` | float | 0.0 | 0.0 - 0.001 | Applies a random perspective transformation to the image, enhancing the model's ability to understand objects in 3D space. | Use very small values (0.0001-0.0005) to maintain realistic object appearances; higher values distort objects excessively. |
| `flipud` | float | 0.0 | 0.0 - 1.0 | Flips the image upside down with the specified probability, increasing the data variability without affecting the object's characteristics. | Generally keep at 0.0; only use (0.3-0.5) for orientation-invariant objects like aerial imagery or symmetric objects. |
| `fliplr` | float | 0.5 | 0.0 - 1.0 | Flips the image left to right with the specified probability, useful for learning symmetrical objects and increasing dataset diversity. | Set to 0.5 for most datasets; reduce to 0.0 for directional objects (text, arrows); essential for doubling effective dataset size. |
| `bgr` | float | 0.0 | 0.0 - 1.0 | Flips the image channels from RGB to BGR with the specified probability, useful for increasing robustness to incorrect channel ordering. | Usually keep at 0.0; set to 0.1-0.3 only when developing models that might be deployed across different image processing pipelines. |
| `mosaic` | float | 1.0 | 0.0 - 1.0 | Combines four training images into one, simulating different scene compositions and object interactions. | Set to 0.7-0.9 for small object detection tasks; reduce to 0.5-0.6 for datasets with thin or fine features (cracks, wires); disable (0.0) in the final few epochs. |
| `mixup` | float | 0.0 | 0.0 - 1.0 | Blends two images and their labels, creating a composite image. Enhances the model's ability to generalize by introducing label noise and visual variability. | Use 0.1-0.2 for small datasets; 0.3-0.4 for complex scenes; avoid or keep very low (0.05) for fine-detail detection tasks. |
| `cutmix` | float | 0.0 | 0.0 - 1.0 | Combines portions of two images, creating a partial blend while maintaining distinct regions. | Set to 0.2-0.3 for classification tasks; lower (0.1) for detection of small objects; higher (0.4-0.5) for robust multi-class detection. |
| `copy_paste` | float | 0.0 | 0.0 - 1.0 | *Segmentation only*. Copies and pastes objects across images to increase object instances. | Use 0.2-0.4 for rare classes with few samples; increase to 0.5-0.7 for extreme class imbalance; critical for improving recall of uncommon objects. |
| `copy_paste_mode` | str | flip | - | *Segmentation only*. Specifies the `copy-paste` strategy to use. Options include `'flip'` and `'mixup'`. | Use 'flip' for most segmentation tasks; try 'mixup' when diversity in object appearance is more important than position. |
| `auto_augment` | str | randaugment | - | *Classification only*. Applies a predefined augmentation policy to enhance model performance through visual diversity. | 'randaugment' works well for general tasks; try 'autoaugment' for known datasets (CIFAR, ImageNet); 'augmix' for highest robustness to distribution shifts. |
| `erasing` | float | 0.4 | 0.0 - 0.9 | *Classification only*. Randomly erases regions of the image during training to encourage the model to focus on less obvious features. | Set to 0.3-0.5 for general classification; increase to 0.6-0.8 for objects with distinctive parts that should not be relied on exclusively. |

## Recommended Parameters

### For General Object Detection
- `mosaic=0.8` - Strong but not overwhelming mosaic augmentation
- `mixup=0.1` - Light blending to maintain feature clarity
- `degrees=10.0` - Moderate rotation for orientation robustness
- `fliplr=0.5` - Standard horizontal flipping
- `translate=0.1` - Slight position variation
- `scale=0.5` - Balanced size variation
- `pretrained=True` - Use transfer learning from COCO dataset
- `batch=16` (or highest that fits on GPU)
- `imgsz=640`
- `epochs=100`
- `patience=50`
- `lr0=0.002`, `lrf=0.01`
- `cos_lr=True`
- `close_mosaic=10`

### For Small Object Detection (e.g., defects, cracks)
- `mosaic=0.6` - Lower mosaic to prevent tiny objects from disappearing
- `mixup=0.05` - Very light blending to preserve small features
- `degrees=5.0` - Minimal rotation
- `scale=0.6` - Emphasize varying sizes
- `hsv_v=0.5` - Enhanced brightness variation
- `copy_paste=0.3` - Increase instances of rare defects
- `imgsz=1280` (larger input size)
- `batch=8` (accommodate larger images)
- `box=10.0` (emphasize localization accuracy)
- `iou=0.35` (more lenient matching criteria)
- `epochs=150` (longer training)
- `close_mosaic=15` (more stabilization epochs)

### For Robust Real-World Deployment
- `hsv_h=0.02, hsv_s=0.7, hsv_v=0.4` - Comprehensive color variation
- `mosaic=0.7` - Balanced mosaic augmentation
- `mixup=0.2` - Moderate blending for robustness
- `translate=0.2` - Increased position variation
- `perspective=0.0003` - Slight perspective change
- `degrees=15.0` - Stronger rotation for viewpoint invariance
- Final training phase after main training:
  - `lr0=0.0001` (very low learning rate)
  - `epochs=20` (short fine-tuning)
  - `rect=True` (rectangular inference optimization)
  - `multi_scale=True` (robust to different image sizes)
  - `batch=4` (smallest stable batch size for best generalization)

### For Limited Dataset Training
- `mosaic=0.9` - Heavy mosaic to maximize data combination
- `mixup=0.2` - Moderate blending for artificial diversity
- `cutmix=0.2` - Additional variation technique
- `copy_paste=0.4` - Replicate scarce objects
- `fliplr=0.5, scale=0.7` - Maximize effective data variation